src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -367,10 +386,48 @@ private void ProcessSettingsFrame(FrameHeader frameHeader)
                     throw new Http2ProtocolException(Http2ProtocolErrorCode.FrameSizeError);
                 }
 
-                // Just eat settings for now
+                // Parse settings and process the ones we care about.
+                ReadOnlySpan<byte> settings = _incomingBuffer.ActiveSpan.Slice(0, frameHeader.Length);
+                while (settings.Length > 0)
+                {
+                    Debug.Assert((settings.Length % 6) == 0);
+
+                    ushort settingId = BinaryPrimitives.ReadUInt16BigEndian(settings);
+                    settings = settings.Slice(2);
+                    uint settingValue = BinaryPrimitives.ReadUInt32BigEndian(settings);
+                    settings = settings.Slice(4);
+
+                    switch ((SettingId)settingId)
+                    {
+                        case SettingId.MaxConcurrentStreams:
+                            // ISSUE 31296: Handle SETTINGS_MAX_CONCURRENT_STREAMS.
+                            break;
+
+                        case SettingId.InitialWindowSize:
+                            if (settingValue > 0x7FFFFFFF)
+                            {
+                                throw new Http2ProtocolException(Http2ProtocolErrorCode.FlowControlError);
+                            }
+
+                            // ISSUE 34059: Handle SETTINGS_MAX_CONCURRENT_STREAMS.
```

Comment:
Nit: Issue #34059 is actually "Respect SETTINGS_INITIAL_WINDOW_SIZE"

Reply:
Thanks for catching, will fix